README.md
# WordPress

This is a WordPress repository configured to run on the [Pantheon platform](https://pantheon.io).

Pantheon is website platform optimized and configured to run high performance sites with an amazing developer workflow. There is built-in support for features such as Varnish, Redis, Apache Solr, New Relic, Nginx, PHP-FPM, MySQL, PhantomJS and more. 

## Getting Started

### 1. Spin-up a site

If you do not yet have a Pantheon account, you can create one for free. Once you've verified your email address, you will be able to add sites from your dashboard. Choose "WordPress" to use this distribution.

### 2. Load up the site

When the spin-up process is complete, you will be redirected to the site's dashboard. Click on the link under the site's name to access the Dev environment.

![alt](http://i.imgur.com/2wjCj9j.png?1, '')

### 3. Run the WordPress installer

How about the WordPress database config screen? No need to worry about database connection information as that is taken care of in the background. The only step that you need to complete is the site information and the installation process will be complete.

We will post more information about how this works but we recommend developers take a look at `wp-config.php` to get an understanding.

![alt](http://i.imgur.com/4EOcqYN.png, '')

If you would like to keep a separate set of configuration for local development, you can use a file called `wp-config-local.php`, which is already in our .gitignore file.

### 4. Enjoy!

![alt](http://i.imgur.com/fzIeQBP.png, '')


#### Setting up Laradock
 
  - Clone Laradock to root of project `git clone https://github.com/Laradock/laradock.git`
  - cd to laradock and copy the env file, `cp env-example .env`
  - Edit the `.env` file and update the following
    - `COMPOSE_PROJECT_NAME=wellgood`
    - `WORKSPACE_INSTALL_WP_CLI=true`

  - Create a new file on root, `wp-config-local.php`

    ```
    <?php
      if (php_sapi_name() == 'cli') {
        define( 'DB_HOST', '127.0.0.1' );
      }
      else {
        define( 'DB_HOST', 'mariadb' );
      }
      define('DB_NAME', 'wellgood');
      define('DB_USER', 'root');
      define('DB_PASSWORD', 'root');
      define('DB_CHARSET',       'utf8');
      define('DB_COLLATE',       '');
      define('AUTH_KEY',         'put your unique phrase here');
      define('SECURE_AUTH_KEY',  'put your unique phrase here');
      define('LOGGED_IN_KEY',    'put your unique phrase here');
      define('NONCE_KEY',        'put your unique phrase here');
      define('AUTH_SALT',        'put your unique phrase here');
      define('SECURE_AUTH_SALT', 'put your unique phrase here');
      define('LOGGED_IN_SALT',   'put your unique phrase here');
      define('NONCE_SALT',       'put your unique phrase here');

      define( 'WP_HOME', 'https://well-good.lndo.site/' );
      define( 'WP_SITEURL', 'https://well-good.lndo.site/' );
    ```

  - Edit laradock/nginx/sites/default.conf

   ```
    server {
        listen 80;
        listen [::]:80;

        # For https
        listen 443 ssl;
        listen [::]:443 ssl ipv6only=on;
        ssl_certificate /etc/nginx/ssl/default.crt;
        ssl_certificate_key /etc/nginx/ssl/default.key;

        server_name well-good.lndo.site;
        root /var/www;
        index index.php index.html index.htm;

        location / {
            try_files $uri $uri/ /index.php$is_args$args;
        }

        location ~ \.php$ {
            try_files $uri /index.php =404;
            fastcgi_pass php-upstream;
            fastcgi_index index.php;
            fastcgi_buffers 16 16k;
            fastcgi_buffer_size 32k;
            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
            #fixes timeouts
            fastcgi_read_timeout 600;
            include fastcgi_params;
        }

        location ~ /\.ht {
            deny all;
        }

        location /.well-known/acme-challenge/ {
            root /var/www/letsencrypt/;
            log_not_found off;
        }

        error_log /var/log/nginx/app_error.log;
        access_log /var/log/nginx/app_access.log;
    }

   ```
  - Edit laradock/php-fpm/php7.3.ini and set `memory_limit = 768M`

  - Edit laradock/docker-compose.yml and **comment** out the following
    ```
      # - "${WORKSPACE_BROWSERSYNC_HOST_PORT}:3000"
      # - "${WORKSPACE_BROWSERSYNC_UI_HOST_PORT}:3001"
    ```

  - Edit laradock/docker-compose.yml and add the following under **enviroment** on the `php-fpm` service
    ```
      - LANDO=ON
      - PANTHEON_ENVIRONMENT=dev
    ```

  - Edit /etc/hosts and add the following
    ```
    127.0.0.1 well-good.lndo.site
    ```

  - Start the docker with the first start might take 15min as it installs all the images
  ```
    docker-compose up -d nginx mariadb workspace
  ```
  - To Stop the docker image
  ```
  docker-compose stop
  ```
  
  - Import/Create DB using SequelPRO, Download a backup [here](https://drive.google.com/file/d/1xHrL2lCaOQF7uQCmY-ViGK9WHLoEn8cL/view?usp=sharing)
  ```
  Connection Settings:
  Host: 127.0.0.1
  Username: root
  Password: root
  ```
 - Select DB from the Dropdown to `wellgood`
 - Import SQL Dump

#### Reduce local DB size
  ```
  wp post delete $(wp post list --post_type=post --format=ids --year=2009) --force && wp post delete $(wp post list --post_type=post --format=ids --year=2010) --force && wp post delete $(wp post list --post_type=post --format=ids --year=2011) --force && wp post delete $(wp post list --post_type=post --format=ids --year=2012) --force && wp post delete $(wp post list --post_type=post --format=ids --year=2013) --force && wp post delete $(wp post list --post_type=post --format=ids --year=2014) --force && wp post delete $(wp post list --post_type=post --format=ids --year=2015) --force && wp post delete $(wp post list --post_type=post --format=ids --year=2016) --force && wp post delete $(wp post list --post_type=post --format=ids --year=2017) --force && wp post delete $(wp post list --format=ids  --post_type=revision --year=2018) --force && wp post delete $(wp post list --format=ids  --post_type=revision --year=2019) --force && wp post delete $(wp post list --format=ids  --post_type=revision --year=2020) --force
  ```

CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

## 17.1.3 - 2020-12-01
### CHANGED:
- Fix z-index of social share buttons
- Sqaure Email Capture XL Position
- Update youtube video structured data for urls

## 17.1.2 - 2020-12-01
### CHANGED:
- Trends 2021 spotlight bugfix

## 17.1.1 - 2020-12-01
### CHANGED:
- Trends 2021 QA
- Fix injector php error

## 17.1.0 - 2020-11-30
### CHANGED:
- Gutenberg comp: newsleter signup
- Trends 2021 QA
- Trends 2021 > Add image placeholders
- Trends 2021 > Improve touch performance
- Custom image compressor wg_resize
- Trends 2021 > Loading improvements

## 17.0.5 - 2020-11-27
### CHANGED:
- Preformance improvements Trends 2021

## 17.0.4 - 2020-11-26
### CHANGED:
- Critical CSS
## 17.0.3 - 2020-11-26
### CHANGED:
- Fix Trends 2021 slideshow figure max-width
- Fix empty vars on header.php

## 17.0.2 - 2020-11-26
### CHANGED:
- Fix reset-p

## 17.0.1 - 2020-11-25
### CHANGED:
- Fix localize_theme
- Fix body text p reset

## 17.0.0 - 2020-11-25
### CHANGED:
- Recaptcha on Signup forms
- Skimlinks Toggle
- Youtube Regex updare for ID
- Upgrade to Node 10 on Jenkins Pipeline
- Gutenberg Editor
- Trends 2021
- Decouple Rich Tag bundle from Main 2020

## 16.4.2 - 2020-11-13
### CHANGED:
- 3 Col row type style
- Add 'Paid Content' label to summer theme
- Update 'Paid Content' on article cards
- Update sponsor vars
- Remove featured content when branded

## 16.4.1 - 2020-11-10
### CHANGED:
- Process scripts/styles
- Typo sizes

## 16.4.0 - 2020-11-10
### CHANGED:
- Process scripts/styles
- 99 Days of Summer CSS Update
- Gift Guide 2021 / V2 Template

## 16.3.4 - 2020-11-05
### CHANGED:
- Fix dropcap in tailwind

## 16.3.3 - 2020-11-05
### CHANGED:
- Tailwind Version
- Fix broken disclaimer
- Fix typography responsive
- Fix typography extends

## 16.3.2 - 2020-10-30
### CHANGED:
- Remove franchise blurb text utility
- Fix drop-cap

## 16.3.1 - 2020-10-30
### CHANGED:
- Update franchise logo size specs

## 16.3.0 - 2020-10-29
### CHANGED:
- Disable Narrativ on Posts
- Fix WYSIWYG in Franchise blurb sponsor text
- Fix parsely verify_title_case date value for Rich Tag page
- Tablet breakpoint update
- Typography respnsive utilities
- pipeline: RO during MySQL replacement
- Styleguide critical + better layout
- Update styleguide naming conventions
- New Typography Styles

## 16.2.3 - 2020-10-23
### CHANGED:
- Remove partner metnion tracking pixel field

## 16.2.2 - 2020-10-19
### CHANGED:
- Add partner tracking pixel
- Fix datalayer

## 16.2.1 - 2020-10-16
### CHANGED:
- Process scripts/styles
- Empty Text Input Fix

## 16.2.0 - 2020-10-15
### CHANGED:
- Process scripts/styles
- Change Label
- Instagram vertical spacin
- Email Capture Close Btn
- Post Share ADA Update
- Email Capture ADA Updates
- Story Link
- ADA Aria Collapse
- Text Input Update

## 16.1.2 - 2020-10-08
### CHANGED:
- Process scripts/styles
- Post Container Max width for Ads - 730px

## 16.1.1 - 2020-10-07
### CHANGED:
- Process scripts/styles
- Process scripts/styles
- Circle EMail Widget Close Btn

## 16.1.0 - 2020-10-06
### CHANGED:
- Add connatix outstream ad
- Add email capture v2
- Update package-lock

## 16.0.6 - 2020-10-04
### CHANGED:
- ADREADY Reset on Load

## 16.0.5 - 2020-10-02
### CHANGED:
- Fix hero data
- Add fallback for feature image in hero

## 16.0.4 - 2020-10-03
### CHANGED:
- Process scripts/styles

## 16.0.3 - 2020-09-30
### CHANGED:
- Fix recipe template
- Fix ._reset

## 16.0.2 - 2020-09-29
### CHANGED:
- Remove clear right from ad

## 16.0.1 - 2020-09-29
### CHANGED:
- Fix fullscreen template

## 16.0.0 - 2020-09-29
### CHANGED:
- Remove _hidden from reset.scss
- Bugfix/youtube api w
- Update tailwind purge
- Update tailwind whitelist
- Add featured content injection logic
- Add featured content promo
- Update infinite scroll call
- Fix get_module on Infinite_Scroll
- Update article template
- Partner in Feed

## 15.2.2 - 2020-09-17
### CHANGED:
- Process scripts/styles
- Fix amp analytics

## 15.2.1 - 2020-09-04
### CHANGED:
- Add link to sponsor

## 15.2.0 - 2020-09-01
### CHANGED:
- Process scripts/styles
- Image links script

## 15.1.0 - 2020-08-31
### CHANGED:
- Process scripts/styles
- Infite Scroll Image links
- Title position and z0index
- XML Escape
- Escape entiries

## 15.0.1 - 2020-08-26
### CHANGED:
- Update default hero settings

## 15.0.0 - 2020-08-26
### CHANGED:
- Update webpack
- New article hero

## 14.13.0 - 2020-08-26
### CHANGED:
- Process scripts/styles
- New tab Checkbox
- Link for Label
- Metafield info
- Story Label
- remove typography changes
- .theme-main-2020 p
- Add file-get-contents
- remove 2020 theme from summer
- Fix base-button-play
- Webpack temporary improvement (added Vendor)

## 14.12.2 - 2020-08-20
### CHANGED:
- Clean up injection logic
- Move native ad
- Update amp ad json

## 14.12.1 - 2020-08-14
### CHANGED:
- Process scripts/styles
- Header Search
- Firefox header fix
- Zindex and scrolling on seatrch
- header QA
- Offset position
- Text Decoration

## 14.12.0 - 2020-08-14
### CHANGED:
- Process scripts/styles
- Zindex and scrolling on seatrch
- header QA
- Offset position
- Text Decoration
- text decoration style

## 14.11.2 - 2020-08-11
### CHANGED:
- Fix headings margin no-legacy

## 14.11.1 - 2020-08-11
### CHANGED:
- Add missing critical css assets

## 14.11.0 - 2020-08-11
### CHANGED:
- Add social_share to rich tag

## 14.10.4 - 2020-08-11
### CHANGED:
- Remove description from featured image on post card
- Fix array_key_exists on image

## 14.10.3 - 2020-08-11
### CHANGED:
- Critical
- Process scripts/styles
- Fix copy-clipboard
- Add copy-clipboard to main bundle
- v14.10.2: PATCH RELEASE
- Process scripts/styles
- Fix and remove social share
- Fix social share + mixins
- v14.10.1: PATCH RELEASE
- Remove legacy assets
- v14.10.0: MINOR RELEASE
- Process scripts/styles
- Finish lazy loading image 2020
- Social share post card 2020 + copy clipboard
- Move transition mixins to main bundle

wp-config.php
<?php

if(!isset($_SERVER['STG_HOST']) && isset($_SERVER['_']) && strpos($_SERVER['_'], 'MAMP')){
  require_once(dirname(__FILE__) . '/wp-config-mamp.php');
}


/**
 * Query monitor config
 */
require_once(dirname(__FILE__) . '/qm-config.php');

/**
 * Redirects naked url and subdomains to www.
 */
if((!isset($_SERVER['LANDO']) || !$_SERVER['LANDO']) && substr( $_SERVER['HTTP_HOST'], 0, 4 ) !== 'www.'){
  $redirect_subdomains = array(
    'archive',
    'summer'
  ); // Specific subdomains to be redirected
  $host_parts = (array) explode('.',$_SERVER['HTTP_HOST']);
  $redirect = false;
  if(sizeof($host_parts) === 2){
    /**
    * Redirect naked urls
    */
    array_unshift($host_parts, 'www');
    $redirect = true;
  } else if(sizeof($host_parts) > 2 && in_array($host_parts[0],$redirect_subdomains)) {
    /**
     * Redirect subdomains '$redirect_subdomains'
     */
    $host_parts[0] = 'www';
    $redirect = true;
  }
  /**
  * 301 redirect
  */
  if($redirect){
    $http_host = implode ( '.' , $host_parts );
    $redirect = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http") . "://{$http_host}{$_SERVER['REQUEST_URI']}";
    http_response_code(301); 
    header("Location: {$redirect}"); 
    exit();
  }
}

/**
 * This config file is yours to hack on. It will work out of the box on Pantheon
 * but you may find there are a lot of neat tricks to be used here.
 *
 * See our documentation for more details:
 *
 * https://pantheon.io/docs
 */

/**
 * Local configuration information.
 *
 * If you are working in a local/desktop development environment and want to
 * keep your config separate, we recommend using a 'wp-config-local.php' file,
 * which you should also make sure you .gitignore.
 */
if (file_exists(dirname(__FILE__) . '/wp-config-local.php') && !isset($_ENV['PANTHEON_ENVIRONMENT'])):
  # IMPORTANT: ensure your local config does not include wp-settings.php
  require_once(dirname(__FILE__) . '/wp-config-local.php');

/**
 * Pantheon platform settings. Everything you need should already be set.
 */
else:
  if (isset($_ENV['PANTHEON_ENVIRONMENT'])):
    // ** MySQL settings - included in the Pantheon Environment ** //
    /** The name of the database for WordPress */
    define('DB_NAME', $_ENV['DB_NAME']);

    /** MySQL database username */
    define('DB_USER', $_ENV['DB_USER']);

    /** MySQL database password */
    define('DB_PASSWORD', $_ENV['DB_PASSWORD']);

    /** MySQL hostname; on Pantheon this includes a specific port number. */
    define('DB_HOST', $_ENV['DB_HOST'] . ':' . $_ENV['DB_PORT']);

    /** Database Charset to use in creating database tables. */
    define('DB_CHARSET', 'utf8');

    /** The Database Collate type. Don't change this if in doubt. */
    define('DB_COLLATE', '');

    /**#@+
     * Authentication Unique Keys and Salts.
     *
     * Change these to different unique phrases!
     * You can generate these using the {@link https://api.wordpress.org/secret-key/1.1/salt/ WordPress.org secret-key service}
     * You can change these at any point in time to invalidate all existing cookies. This will force all users to have to log in again.
     *
     * Pantheon sets these values for you also. If you want to shuffle them you
     * must contact support: https://pantheon.io/docs/getting-support 
     *
     * @since 2.6.0
     */
    define('AUTH_KEY',         $_ENV['AUTH_KEY']);
    define('SECURE_AUTH_KEY',  $_ENV['SECURE_AUTH_KEY']);
    define('LOGGED_IN_KEY',    $_ENV['LOGGED_IN_KEY']);
    define('NONCE_KEY',        $_ENV['NONCE_KEY']);
    define('AUTH_SALT',        $_ENV['AUTH_SALT']);
    define('SECURE_AUTH_SALT', $_ENV['SECURE_AUTH_SALT']);
    define('LOGGED_IN_SALT',   $_ENV['LOGGED_IN_SALT']);
    define('NONCE_SALT',       $_ENV['NONCE_SALT']);
    /**#@-*/

    /** A couple extra tweaks to help things run well on Pantheon. **/
    if (isset($_SERVER['HTTP_HOST'])) {
        // HTTP is still the default scheme for now. 
        $scheme = 'http';
        // If we have detected that the end use is HTTPS, make sure we pass that
        // through here, so <img> tags and the like don't generate mixed-mode
        // content warnings.
        if (isset($_SERVER['HTTP_USER_AGENT_HTTPS']) && $_SERVER['HTTP_USER_AGENT_HTTPS'] == 'ON') {
            $scheme = 'https';
            $_SERVER['HTTPS'] = 'on';
        }
        define('WP_HOME', $scheme . '://' . $_SERVER['HTTP_HOST']);
        define('WP_SITEURL', $scheme . '://' . $_SERVER['HTTP_HOST']);
    }
    // Don't show deprecations; useful under PHP 5.5
    error_reporting(E_ALL ^ E_DEPRECATED);
    /** Define appropriate location for default tmp directory on Pantheon */
    define('WP_TEMP_DIR', $_SERVER['HOME'] .'/tmp');

    // FS writes aren't permitted in test or live, so we should let WordPress know to disable relevant UI
    if ( in_array( $_ENV['PANTHEON_ENVIRONMENT'], array( 'test', 'live' ) ) && ! defined( 'DISALLOW_FILE_MODS' ) ) :
        define( 'DISALLOW_FILE_MODS', true );
    endif;

  else:
    /**
     * This block will be executed if you have NO wp-config-local.php and you
     * are NOT running on Pantheon. Insert alternate config here if necessary.
     *
     * If you are only running on Pantheon, you can ignore this block.
     */
    define('DB_NAME',          'database_name');
    define('DB_USER',          'database_username');
    define('DB_PASSWORD',      'database_password');
    define('DB_HOST',          'database_host');
    define('DB_CHARSET',       'utf8');
    define('DB_COLLATE',       '');
    define('AUTH_KEY',         'put your unique phrase here');
    define('SECURE_AUTH_KEY',  'put your unique phrase here');
    define('LOGGED_IN_KEY',    'put your unique phrase here');
    define('NONCE_KEY',        'put your unique phrase here');
    define('AUTH_SALT',        'put your unique phrase here');
    define('SECURE_AUTH_SALT', 'put your unique phrase here');
    define('LOGGED_IN_SALT',   'put your unique phrase here');
    define('NONCE_SALT',       'put your unique phrase here');
  endif;
endif;

/** Standard wp-config.php stuff from here on down. **/

/**
 * Disable default wp cron
 */
define('DISABLE_WP_CRON', true);

/**
 * WordPress Database Table prefix.
 *
 * You can have multiple installations in one database if you give each a unique
 * prefix. Only numbers, letters, and underscores please!
 */
$table_prefix = 'wp_';

/**
 * WordPress Localized Language, defaults to English.
 *
 * Change this to localize WordPress. A corresponding MO file for the chosen
 * language must be installed to wp-content/languages. For example, install
 * de_DE.mo to wp-content/languages and set WPLANG to 'de_DE' to enable German
 * language support.
 */
define('WPLANG', '');

/**
 * For developers: WordPress debugging mode.
 *
 * Change this to true to enable the display of notices during development.
 * It is strongly recommended that plugin and theme developers use WP_DEBUG
 * in their development environments.
 *
 * You may want to examine $_ENV['PANTHEON_ENVIRONMENT'] to set this to be
 * "true" in dev, but false in test and live.
 */
define( 'WP_DEBUG', true );

if ( !empty( $_SERVER['PANTHEON_ENVIRONMENT'] ) && ( "cli" !== php_sapi_name() ) ) {
  // set debug to true in all environments except live
  if ( "live" !== $_SERVER['PANTHEON_ENVIRONMENT'] && !defined( 'WP_DEBUG' ) ) {
    define( 'WP_DEBUG', true );
    define( 'WP_DEBUG_LOG', true );
  }

  // upgrade to https if headers forwarded from CDN like Cloudflare and terminating https
  if ( isset( $_SERVER['HTTP_X_FORWARDED_PROTO'] ) 
    && 'https' == strtolower( $_SERVER['HTTP_X_FORWARDED_PROTO'] ) ) {
    $_SERVER['HTTPS'] = 'on';
  }

  // some services use SERVER_NAME, which is unreliable here. This seems to fix those issues.
  $_SERVER['SERVER_NAME'] = $_SERVER['HTTP_HOST'];
  $_SERVER['SERVER_PORT'] = ( 
    isset( $_SERVER['HTTP_X_SSL'] ) && 'ON' === strtoupper( $_SERVER['HTTP_X_SSL'] ) ||
    @$_SERVER['HTTPS'] === 'on'
  ) ? 443 : 80;
}
if ( ! defined( 'WP_DEBUG' ) ) {
    define('WP_DEBUG', false);
}

/* That's all, stop editing! Happy Pressing. */



/** Absolute path to the WordPress directory. */
if ( !defined('ABSPATH') )
  define('ABSPATH', dirname(__FILE__) . '/');
  
/**
 * Inject redirection
 */
$redirection = dirname(__FILE__). '/wp-content/plugins/wg-redirections/wp-config.php';
if(file_exists($redirection)) require_once $redirection;

/** Sets up WordPress vars and included files. */
require_once(ABSPATH . 'wp-settings.php');
